test/unit/datasette.test.js

```diff
@@ -0,0 +1,52 @@
+import datasette, { formatData } from '../../src/services/datasette.js'
+import axios from 'axios'
+import { vi, describe, beforeEach, afterEach, it, expect } from 'vitest'
+
+describe('datasette', () => {
+  beforeEach(() => {
+    vi.spyOn(axios, 'get').mockResolvedValue({
+      data: {
+        columns: ['column1', 'column2'],
+        rows: [
+          ['value1', 'value2'],
+          ['value3', 'value4']
+        ]
+      }
+    })
+  })
+
+  afterEach(() => {
+    vi.mocked(axios.get).mockReset()
+  })
+
+  it('runs a SQL query and returns the results', async () => {
+    const query = 'SELECT * FROM table_name'
+    const result = await datasette.runQuery(query)
+
+    expect(result).toHaveProperty('columns')
```

Comment:
That's a pretty roundabout way of saying:
 
```
expect(result).toEqual({
        columns: ['column1', 'column2'],
        formattedData: [
          { column1: 'value1', column2: 'value2' },
          { column1: 'value3', column2: 'value4' }
        ],
        rows: [
          ['value1', 'value2'],
          ['value3', 'value4']
        ]
    })
  })
```
Which makes it pretty clear what the expected shape of the data is.

It applies to other tests (e.g. the `formatData` one)

Reply:
might be easier to work out exactly what part of this test is broken. happy to change if you feel strongly about it though

Reply:
Vitest already gives you nice diff in the output, so not sure if it can get easier than that: 
![image](https://github.com/user-attachments/assets/5209bedb-3f09-4220-9b38-c2c935ad62aa)

Reply:
coolio, have updated